app/src/main/java/org/simple/clinic/summary/compose/StatinNudgeView.kt

```diff
@@ -113,48 +172,105 @@ fun RiskProgressBar() {
 
   val indicatorColor = Color(0xFF2F363D)
 
-  Box(
+  BoxWithConstraints(
       modifier = Modifier
           .fillMaxWidth()
           .height(14.dp)
           .drawWithContent {
             drawContent()
 
-            val widthPerSegment = size.width / riskColors.size
-
             drawLine(
                 color = indicatorColor,
-                start = Offset(2 * widthPerSegment, 0f),
-                end = Offset(2 * widthPerSegment, size.height),
+                start = Offset(startOffset, 0f),
+                end = Offset(startOffset, size.height),
                 strokeWidth = 2.dp.toPx()
             )
             drawLine(
                 color = indicatorColor,
-                start = Offset(size.width, 0f),
-                end = Offset(size.width, size.height),
+                start = Offset(endOffset, 0f),
+                end = Offset(endOffset, size.height),
                 strokeWidth = 2.dp.toPx()
             )
           },
       contentAlignment = Alignment.Center,
   ) {
+    val totalSegments = riskColors.size
+    val segmentWidthPx = constraints.maxWidth.toFloat() / totalSegments
+
     Row(
         modifier = Modifier
             .fillMaxWidth()
             .height(4.dp)
             .clip(RoundedCornerShape(50))
     ) {
-      riskColors.forEach { color ->
+      riskColors.forEachIndexed { index, color ->
+        val segmentStartPx = index * segmentWidthPx
+        val segmentEndPx = (index + 1) * segmentWidthPx
+
         Box(
             modifier = Modifier
                 .weight(1f)
                 .fillMaxHeight()
-                .background(color),
+                .drawWithContent {
+                  drawRect(color.copy(alpha = 0.5f))
+
+                  val visibleStart = maxOf(segmentStartPx, startOffset)
+                  val visibleEnd = minOf(segmentEndPx, endOffset)
+
+                  if (visibleStart < visibleEnd) {
+                    drawRect(
+                        color = color.copy(alpha = 1.0f),
+                        topLeft = Offset(x = visibleStart - segmentStartPx, y = 0f),
+                        size = Size(
+                            width = visibleEnd - visibleStart,
+                            height = size.height
+                        )
+                    )
+                  }
+                }
```

Comment:
I am just curious, is it possible to replace this `Row` and `Box` usage and directly draw in canvas? Since we are already doing it in the `Box`?

Reply:
I tried using canvas, and clipping and size of risk progress bar were not working properly. So i had to implement it using box and row.